sympy/integrals/manualintegrate.py

```diff
@@ -678,24 +678,26 @@ def trig_rule(integral):
             func = 'cos'
 
         return TrigRule(func, arg, integrand, symbol)
-
-    if integrand == sympy.sec(symbol)**2:
-        return TrigRule('sec**2', symbol, integrand, symbol)
-    elif integrand == sympy.csc(symbol)**2:
-        return TrigRule('csc**2', symbol, integrand, symbol)
+    if (len(integrand.args) <= 2 and isinstance(integrand.args[0], (sympy.sec, sympy.csc)) or isinstance(integrand, (sympy.sec, sympy.csc))):
+        if isinstance(integrand, sympy.sec):
+            return TrigRule('sec', integrand.args[0], integrand, symbol)
+        elif isinstance(integrand, sympy.csc):
+            return TrigRule('csc', integrand.args[0], integrand, symbol)
+
+        if (integrand.args[0] == symbol):
```

Comment:
Please add some tests that covers this branch of the code.

Reply:
Please review it again

Reply:
Actually all the difference according to the codecov report are in those files which are not linked to the files are worked on . And I have not changed any other file

Reply:
Yes, the Codecov issue is a bit annoying. However, the interesting part is: 

The diff coverage is 76.19%.

Which means that out of the lines you have changed, about one quarter is not tested, namely the branch I asked to be tested. And as you can see at https://codecov.io/gh/sympy/sympy/pull/16583/diff it is still not tested. (Basically, either you know how to reach it and can therefore add a test for it, if someone else should figure it out later it may be close to impossible. If you cannot figure it out, maybe it doesn't have to be there.)